proposals/storage-and-retrieval-bots.md

```diff
@@ -0,0 +1,168 @@
+	"failedAt":"ClientEventProviderCanceled",  // failure event
+	"eventList":
+		[
+			"Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)",
+			"Recv: 0 B, Paid 0 FIL, ClientEventDealProposed (DealStatusWaitForAcceptance)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventDealAccepted (DealStatusAccepted)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventPaymentChannelAddingFunds (DealStatusPaymentChannelAllocatingLane)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventLaneAllocated (DealStatusOngoing)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventProviderCancelled (DealStatusCancelling)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventDataTransferError (DealStatusErrored)".
+			"Recv: 0 B, Paid 0 FIL, ClientEventOpen (DealStatusNew)".
+		],
+	"errorMessage":"ERROR: retrieval failed: Retrieve: Retrieval Error: error generated by data transfer: unable to send cancel to channel FSM: normal shutdown of state machine",
+	"tailLog":"....",            // Multiline, from `tail` of `lotus daemon`
+	"storageDealParameters":      // Given to RetrievalBot as input 
+		{
+			"CID":"Qm...",
+			"sha256":"73cb385...",    // independent checksum of data file
+			"sizeInBytes":"12345678".
+			"minderId":"f01924",
+			"verified":true,
+			"fastRetrievalFlag":true,
+			"dealId":"...",
+		}
+		"lotusVersion":"1.5.3-rc2+mainnet+git.9afb5ff94",
+												               // Call API `Filecoin.Version`
+		"datetime":"YYYY-MM-DD_HH:MM:SS",  // when attempt started
+},
+{
+  // ...next deal attempt json blob...
+}
+```
+
+Stdout will contain the results, in json, of each deal attempt.  It is intended to be piped into a log search service like those provided by AWS/GC.
+
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+ - Improve reliability of the network
+ - Enable an ecosystem of miner reputation and ranking systems
+ - Perform the retrieval verification in Slingshot 2.3
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+**Immensely!**
+
+ - We don't currently have enough information about why deals fail to allocate our debugging time and resources correctly.
+
+ - Miner reputation systems enabled by this tool would compliment the protocol-level incentives for miners to "do the right thing" (provide reliable retrieval of previously stored data, successfully complete all storage deals, etc)
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+C = 8
+
+Nothing is certain, but it is very likely that building this tool will at a minimum enable the Filecoin Project to better understand the frequency and causes of deal failures.  
+
+And the ability of this tool to support miner reputation systems can only help increase deals that get routed to reliable miners.
+
+
+## Project definition
+#### Brief plan of attack
```

Comment:
ℹ️  Noting that this project doesn't list
* Unit tests
* Monitoring of ongoing functionality of the bots
* Documentation for usage by external entities

With these, we should expect this project to be at least a 'medium' or to take at least 4 weeks of time.

Reply:
I'm 👍 on tests and docs.

For ongoing monitoring, the [dashboards project](https://github.com/protocol/web3-dev-team/pull/85) might inadvertently solve for that.  If the bots start failing, it will probably be immediately apparent to viewers of the dashboards (e.g., all metrics suddenly go to zero).